src/App.tsx

```diff
@@ -1,27 +1,57 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import * as goodsFilterFunctionAPI from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const buttonClickHandle = async (loadedGoods: Promise<Good[]>) => (
+    setGoods(await loadedGoods)
+  );
```

Comment:
still I can't get what does this function do. Don't name your functions like `handleClickSomewhere`, because it does not give me **any** additional information. I should understand **exactly** what it does by its name.

Reply:
userDefinedCallOnAPI

is that clearer as a variable name? Or do you want the variable to describe the asynchronous function calling for goods?
I'm a bit unsure what you mean here...